script/releases/v1.0.0-slashing/2-queueUpgradeAndUnpause.s.sol

```diff
@@ -0,0 +1,148 @@
+            /// core/
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.avsDirectory()),
+                    impl: address(Env.impl.avsDirectory())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.delegationManager()),
+                    impl: address(Env.impl.delegationManager())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.rewardsCoordinator()),
+                    impl: address(Env.impl.rewardsCoordinator())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.strategyManager()),
+                    impl: address(Env.impl.strategyManager())
+                })
+            })
+            /// pods/
+            .append({
+                to: address(Env.proxy.eigenPod()),
+                data: Encode.upgradeableBeacon.upgradeTo({
+                    newImpl: address(Env.impl.eigenPod())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.eigenPodManager()),
+                    impl: address(Env.impl.eigenPodManager())
+                })
+            })
+            /// strategies/
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.eigenStrategy()),
+                    impl: address(Env.impl.eigenStrategy())
+                })
+            })
+            .append({
+                to: address(Env.proxy.strategyBase()),
+                data: Encode.upgradeableBeacon.upgradeTo({
+                    newImpl: address(Env.impl.strategyBase())
+                })
+            })
+            .append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: address(Env.proxy.strategyFactory()),
+                    impl: address(Env.impl.strategyFactory())
+                })
+            });
+
+        /// Add call to upgrade each pre-longtail strategy instance
+        uint count = Env.instance.strategyBaseTVLLimits_Count();
+        for (uint i = 0; i < count; i++) {
+            address proxyInstance = address(Env.instance.strategyBaseTVLLimits(i));
+
+            executorCalls.append({
+                to: Env.proxyAdmin(),
+                data: Encode.proxyAdmin.upgrade({
+                    proxy: proxyInstance,
+                    impl: address(Env.impl.strategyFactory())
```

Comment:
can we change this to the Strategy factory beacon? That way we never need to special case old strategies

Reply:
See Yash's comment above

Reply:
okay chatted w/ G - the request is to make the instanced strategies proxy through a beacon proxy that points to the beacon used by the rest of our strategies

im tempted to leave this as-is. The existing strategies will still be edge cases in the StrategyFactory because of the blacklist/deploy system the factory uses -- so rather than try and fix this now, maybe we should leave as-is, given timelines are tight.